### ISSUE TITLE
Add a pre-score extension point to the scheduling framework 

### ISSUE THREAD

**Issue by ahg-g** [kubernetes/kubernetes#80336]
**Why is this needed**:
Cases, like even pod spreading, will need to pre-compute some data across all nodes, not just the ones that passed the filter phase. 

/assign
/sig scheduling

**Comment by hex108** [comment]
According to the design in [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/20180409-scheduling-framework.md), it seems that pre-score could be done in [pos-filter](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/20180409-scheduling-framework.md#post-filter).

> Note: Plugins wishing to perform "pre-scoring" work should use the post-filter extension point.

**Comment by draveness** [comment]
The post-filter extension point is currently blocked by the filter extension point, I'll continue to work on that after filter get merged into the codebase.

**Comment by ahg-g** [comment]
ok, sounds good, I thought post-filter plugins will be executed per pod per node, not per pod. If it is invoked once per pod, then that solves the problem.
